### ISSUE TITLE
Fail CI for warnings

### ISSUE THREAD

**Issue by bsipocz** [numpy/numpy-tutorials#154]
Ideally we should handle warnings raised during the tutorials, and fail CI with anything that is unexpected.

**Comment by rossbar** [comment]
Good point! I though we had this covered with the `-W` option for sphinx-build, but it turns out that doesn't extend to warnings raised *during notebook execution*.

**Comment by bsipocz** [comment]
yep, the code content and rendering are handled separately. 

You can assign this to me, though I'm not sure when I get to actually do something about it.

**Comment by rossbar** [comment]
Of course - feel free! I also opened executablebooks/MyST-NB#459 about this as I couldn't find a good way to handle this at the config-level.

**Comment by bsipocz** [comment]
> Of course - feel free

I can't do that ;)

> I also opened https://github.com/executablebooks/MyST-NB/issues/459 about this as I couldn't find a good way to handle this at the config-level.

Ideally, a standard pytest config should do the trick, but f course it doesn't work out of the box :(

**Comment by rossbar** [comment]
I'm going to go ahead and re-open this - #203 fixed the current batch of warnings, but I agree with the original premise that we need something in CI to fail on runtime warnings.

**Comment by rossbar** [comment]
I *believe* this is no longer directly relevant with the JB2 build/execution toolchain, right? Once we tidy things up enough to enable the `--strict` flag (already recorded as a follow-up in #280), that should resolve this issue?

WDYT @bsipocz - I'm happy to keep this open if it's still relevant, though I suspect the discussion isn't anymore after the infrastructure switch!
